src/assets/scripts/client/airport/MapLayerModel.js

```diff
@@ -0,0 +1,118 @@
+import _forEach from 'lodash/forEach';
+import _isEqual from 'lodash/isEqual';
+import _isNumber from 'lodash/isNumber';
+import _map from 'lodash/map';
+import BaseModel from '../base/BaseModel';
+import DynamicPositionModel from '../base/DynamicPositionModel';
+import StaticPositionModel from '../base/StaticPositionModel';
+
+/**
+ * A collection of lines that represent a map layer.
+ *
+ * @class MapLayerModel
+ */
+export default class MapLayerModel extends BaseModel {
+    /**
+     * @for MapLayerModel
+     * @constructor
+     * @param map {object}
+     * @param airportPosition {StaticPositionModel}
+     * @param magneticNorth {number}
+     */
+    constructor(map, airportPosition, magneticNorth) {
+        super();
+
+        if (!map || !airportPosition || !_isNumber(magneticNorth)) {
+            // eslint-disable-next-line max-len
+            throw new TypeError('Invalid parameter, expected map, airportPosition and magneticNorth to be defined');
+        }
+
+        /**
+         * A flag indicating whether the layer should be hidden
+         *
+         * @for MapLayerModel
+         * @property hidden
+         * @type {boolean}
+         * @default false
+         */
+        this.hidden = false;
+
+        /**
+         * List of lat/long coordinates pairs that define a line
+         *
+         * @for MapLayerModel
+         * @property lines
+         * @type {array}
+         * @default []
+         */
+        this.lines = [];
+
+        /**
+         * Name of the map layer.
+         *
+         * @for MapLayerModel
+         * @property name
+         * @type {string}
+         * @default ''
+         */
+        this.name = '';
+
+        return this._init(map, airportPosition, magneticNorth);
+    }
+
+    /**
+     * Initialize the model
+     *
+     * @for MapLayerModel
+     * @method _init
+     * @param map {object}
+     * @param airportPosition {StaticPositionModel}
+     * @param magneticNorth {number}
+     * @private
+     */
+    _init(map, airportPosition, magneticNorth) {
+        this.hidden = map.hidden === true;
+        this.name = map.name;
+        this._buildMapLines(map.lines, airportPosition, magneticNorth);
+
+        return this;
+    }
+
+    /**
+     * Create the array of map lines
+     * @for MapLayerModel
+     * @method _buildMapLines
+     * @param map {object}
+     * @param airportPosition {StaticPositionModel}
+     * @param magneticNorth {number}
+     * @private
+     */
+    _buildMapLines(lines, airportPosition, magneticNorth) {
+        _forEach(lines, (line) => {
```

Comment:
could we use `map` here instead?  probably not because `lines` is an object?

Reply:
The param type was wrong, it should have been an array, so I think `lines` correctly describes the contents
```json
"lines": [
    ["N45.00374744", "W063.74988584", "N45.00715629", "W063.74650952"],
    ["N45.00715629", "W063.74650952", "N45.00715649", "W063.74650992"]
]
```

Reply:
oh, perfect.  then we can use `.map()` here instead of `_forEach`

Reply:
Ah, that's what you meant. I had thought you were referring to the variable name...